pom.xml

```diff
@@ -2366,6 +2366,8 @@
                                     <!-- needs some careful inspection -->
                                     -Xep:GuardedBy:OFF \
                                     -Xep:HidingField:ERROR \
+                                    <!-- some Trino classes are marked as @Immutable but in fact they contain mutable fields and fail the check -->
+                                    -Xep:Immutable:OFF \
```

Comment:
Sounds disturbing.

BTW how does this related to airlift update?

Reply:
It's related to the switch from javax.annotation to jakarta.annotation and errorprone_annotations. Seems that previously the `@Immutable` wasn't checked by the error-prone at all (due to the wrong annotation being used). The annotation right now is picked by error-prone and it fails for some of the cases that are not immutable at all (i.e. they contain mutable fields).

As stated by error-prone rule documentation:

`This check validates that all classes annotated with Error Prone’s @Immutable annotation (com.google.errorprone.annotations.Immutable) are deeply immutable.`

Reply:
I.e. we have an `@Immutable` annotation on the `TypeSignature` which contains `@Immutable` `TypeSignatureParameter` but this class contains `Object` field which is not immutable according to the `error-prone` and thus the rule fails for `TypeSignature`.

Reply:
I guess that disturbing part is that we've been using this annotation but it wasn't enforced at all previously.

Reply:
change comment to `<!-- This check requires list fields to be declared as ImmutableList, which is not in line with project's code style -->`

Reply:
It's not only about List<?> vs ImmutableList<?> so I don't think that this comment is accurate

Reply:
add `e.g.`

it's definitely more accurate than the one suggesting that Trino `@immutable` classes are actually mutable.